### ISSUE TITLE
2019-12-20 tc011 M1SS PTROUGHR2/PTROUGHREND2 ACTUAL TIME with commit 5ca1f6e is missing

### ISSUE THREAD

**Issue by tensfeldt** [tensfeldt/openNCA#132]
parameterset == "PARAMETERLIST"
mct$TIME == "ACTUAL"

```r
      SDEID DOSE1 DOSE2 DOSEC CMAX1 CMAX2  CMAXDN1  CMAXDN2 FLGACCEPTPREDOSE CMIN1 CMIN2 CLAST1 CLAST2 TMAX1  TMAX2 FLGACCEPTTMAX TMIN1
1  70745987   300   300   300  7460  6750 24.86667 22.50000                0  2310  2560   2680   2560 1.517 14.000             1     0
2 130509330   300   300   300  8120  9540 27.06667 31.80000                0  1940  2200   2290   2200 1.500 13.017             1     0
3 246409536   300   300   300  8130 12600 27.10000 42.00000                0   945  2240   2240   2420 0.667 12.667             1     0
4 249265153   300   300   300 12200 12100 40.66667 40.33333                0  1620  1720   2030   1720 1.000 13.000             1     0
   TMIN2 TLAST TLAG CTROUGH1 CTROUGH2 CTROUGHEND1 CTROUGHEND2 PTROUGHR1 PTROUGHR2 PTROUGHREND1 PTROUGHREND2        KEL    KELC0 KELTMLO
1 24.000    24    0     2310       NA        2680          NA  3.229437        NA     2.783582           NA 0.10407221 31122.16      18
2 24.000    24    0     1940       NA        2290          NA  4.185567        NA     3.545852           NA 0.10416094 26895.50      18
3 11.917    24    0      945       NA        2240          NA  8.603175        NA     3.629464           NA 0.09888024 25947.24      16
4 24.000    24    0     1620       NA        2030          NA  7.530864        NA     6.009852           NA 0.13193783 40656.18      16
  KELTMHI KELNOPT    KELRSQ   KELRSQA FLGACCEPTKEL    THALF THALFF LASTTIME FLGACCEPTTAU   AUCALL AUCLASTC AUCLASTDN AUCLAST1 AUCLAST2
1      24       3 0.9999990 0.9999980            1 6.660252      1       24            1 109073.1 85050.91 0.3635771 54349.39 54779.57
2      24       3 0.9995933 0.9991866            1 6.654578      1       24            1 109313.5 89159.40 0.3643782 52409.32 56607.78
3      24       4 0.9990935 0.9986402            1 7.009967      1       24            1 106414.7 95967.10 0.3547157 47539.09 58689.05
4      24       4 0.9993207 0.9989811            1 5.253589      1       24            1 105601.8 92805.77 0.3520061 52762.04 53036.75
  AUCLASTC1 AUCLASTC2 AUCLASTDN1 AUCLASTDN2      AUC1     AUC2     AUC3      AUC4     AUC5     AUC6     AUC7     AUC8     AUC9    AUC10
1  25731.52  30757.34  0.1811646  0.1825986  820.8450 2370.605 4380.260  7707.155 11160.93 23654.42 33745.31 45496.35 54349.39 55443.47
2  28401.43  36453.71  0.1746977  0.1886926 1084.6650 2182.125 4499.805  8432.305 12228.99 24432.60 33924.17 44753.77 52409.32 53368.20
3  35040.58  48241.42  0.1584636  0.1956302  994.8375 3192.557 5815.717  9411.433 12523.92 22401.10 30375.09 40110.63 47539.09 49781.33
4  37940.30  40240.69  0.1758735  0.1767892  682.6500 2464.540 5859.475 11018.396 14951.49 27189.14 36127.20 45885.53 52762.04 54010.04
     AUC11    AUC12    AUC13    AUC14    AUC15    AUC16     AUC17    AUC18 AUCINT1 AUCINT2 AUCINT3 AUCINT4 AUCINT5 AUCINT6 AUCINT7 AUCINT8
1 56709.33 58482.08 61056.47 64126.47 76965.50 87791.91 100112.33 109129.0 0_0.333 0_0.667     0_1 0_1.517     0_2     0_4     0_6     0_9
2 55337.13 58663.88 63023.23 67215.32 80920.03 90580.44 101224.89 109017.1 0_0.433 0_0.667     0_1   0_1.5     0_2     0_4     0_6     0_9
3 53311.71 56870.83 60955.76 64541.98 76697.41 86402.61  97811.52 106228.1 0_0.333 0_0.667     0_1   0_1.5     0_2     0_4 0_6.017     0_9
4 56356.39 60009.40 65190.69 69251.75 81438.83 90126.57  99563.79 105798.8 0_0.333 0_0.667     0_1   0_1.5     0_2     0_4     0_6 0_9.017
   AUCINT9 AUCINT10 AUCINT11 AUCINT12 AUCINT13 AUCINT14 AUCINT15 AUCINT16 AUCINT17 AUCINT18   AUC0_12 AUC12_24  AUC0_48 AUC15_30 AUC12_26
1 0_11.917 0_12.333 0_12.667 0_13.017   0_13.5     0_14     0_16     0_18     0_21     0_24 109142.00 109004.4 263297.9 99699.57 118249.8
2 0_11.917 0_12.333 0_12.667 0_13.017   0_13.5     0_14     0_16     0_18     0_21     0_24 105792.01 112834.9 257445.2 88376.75 120792.7
3 0_11.917 0_12.333 0_12.667     0_13   0_13.5     0_14     0_16     0_18     0_21     0_24  95927.52 116901.9 257204.4 92342.09 125681.3
4 0_11.933 0_12.333 0_12.667     0_13   0_13.5     0_14     0_16     0_18 0_21.017     0_24 105819.89 105431.7 236152.6 73980.55 111468.1
   AUCINFO AUCINFOC AUCINFODN  AUCINFP AUCINFPC AUCINFPDN  AUCTAU1  AUCTAU2 AUCTAUDN1 AUCTAUDN2 AUMCTAU1 AUMCTAU2  MRTLAST MRTLAST1
1 133671.4 111475.3 0.4455715 133675.9 111479.7 0.4455862 54570.99 54779.57 0.1819033 0.1825986 282266.6 945806.0 11.22732 5.144797
2 130434.6 111809.6 0.4347821 130511.6 111886.6 0.4350388 52599.63 56607.78 0.1753321 0.1886926 259696.9 944461.7 10.96689 4.911741
3 130888.8 121331.8 0.4362959 130868.5 121311.5 0.4362284 47752.20 58689.05 0.1591740 0.1956302 234730.5 977473.9 11.34501 4.884015
4 118638.3 106359.8 0.3954609 118590.4 106311.9 0.3953012 52906.71 53036.75 0.1763557 0.1767892 243369.6 868174.0 10.48452 4.579776
  MRTLAST2 MRTEVIFO1 MRTEVIFO2 MRTEVIFP1 MRTEVIFP2 AUCXPCTO AUCXPCTP     CAV1     CAV2     CLFTAU1     CLFTAU2     PTF1      PTF2     PTR1
1 17.26565  442.0418 1790.6028  551.6875 2386.8808 18.40206 18.40476 4547.582 4564.964 0.005497427 0.005476495 1.132470 0.9178605 3.229437
2 16.68432  221.8119  562.1702  273.0663  690.6337 16.19291 16.24236 4383.303 4717.315 0.005703462 0.005299625 1.409896 1.5559699 4.185567
3 16.65662  222.8520  473.1368  262.0371  511.9006 18.69836 18.68577 3979.350 4890.754 0.006282433 0.005111686 1.805571 2.1182828 8.603175
4 16.36975  139.5435  279.2962  169.5875  325.9685 10.98840 10.95244 4408.892 4419.729 0.005670358 0.005656455 2.399696 2.3485603 7.530864
      PTR2    VZFTAU1    VZFTAU2  VZFTAUW1  VZFTAUW2 CONC1 CONC2 CONC3 CONC4 CONC5 CONC6 CONC7 CONC8 CONC9 CONC10 CONC11 CONC12 CONC13
1 2.636719 0.05282319 0.05262206 0.6364240 0.6340008 2.310  2.62  6.66  5.41  7.46  6.85  5.68  4.46  3.42   2.68   2.58   5.00   5.13
2 4.336364 0.05475624 0.05087920 0.7924203 0.7363125 1.940  3.07  6.31  7.61  8.12  7.09  5.21  4.31  2.99   2.29   2.32   9.47   9.54
3 5.625000 0.06353578 0.05169573 0.7568288 0.6157919 0.945  5.03  8.13  7.63  6.77  5.71  4.24  3.68  2.88   2.24   8.54  12.60   8.98
4 7.034884 0.04297750 0.04287213 0.6371757 0.6356134 1.620  2.48  8.19 12.20  8.64  7.14  5.20  3.81  2.72   2.03   4.21   9.84  12.10
  CONC14 CONC15 CONC16 CONC17 CONC18 CONC19 CONCTIME1 CONCTIME2 CONCTIME3 CONCTIME4 CONCTIME5 CONCTIME6 CONCTIME7 CONCTIME8 CONCTIME9
1   5.53   6.75   6.10   4.78   3.50   2.56         0     0.333     0.667         1     1.517         2         4     6.000     9.000
2   8.53   8.24   5.63   4.11   3.04   2.20         0     0.433     0.667         1     1.500         2         4     6.000     9.000
3   7.41   6.94   5.29   4.44   3.23   2.42         0     0.333     0.667         1     1.500         2         4     6.017     9.000
4   8.80   7.48   4.89   3.84   2.51   1.72         0     0.333     0.667         1     1.500         2         4     6.000     9.017
  CONCTIME10 CONCTIME11 CONCTIME12 CONCTIME13 CONCTIME14 CONCTIME15 CONCTIME16 CONCTIME17 CONCTIME18 CONCTIME19 DI1 DI2 TAU1 TAU2 TOLD1
1     11.917     12.333     12.667     13.017       13.5         14         16         18     21.000         24  NA  NA   12   12     0
2     11.917     12.333     12.667     13.017       13.5         14         16         18     21.000         24  NA  NA   12   12     0
3     11.917     12.333     12.667     13.000       13.5         14         16         18     21.000         24  NA  NA   12   12     0
4     11.933     12.333     12.667     13.000       13.5         14         16         18     21.017         24  NA  NA   12   12     0
  TOLD2 TIMEU AMOUNTU DOSE1U DOSE2U VOLUMEU CONCU KELU  CLU     AUCU       AUMCU AUCNORMU AURCU CONCNORMU RATEU VOLUMEWU CLWU    STUDY
1    12    HR      NA     MG     MG       L NG/ML 1/HR L/HR NG.HR/ML NG.HR.HR/ML       NA    NA  NG/ML/MG    NA       NA   NA A0081216
2    12    HR      NA     MG     MG       L NG/ML 1/HR L/HR NG.HR/ML NG.HR.HR/ML       NA    NA  NG/ML/MG    NA       NA   NA A0081216
3    12    HR      NA     MG     MG       L NG/ML 1/HR L/HR NG.HR/ML NG.HR.HR/ML       NA    NA  NG/ML/MG    NA       NA   NA A0081216
4    12    HR      NA     MG     MG       L NG/ML 1/HR L/HR NG.HR/ML NG.HR.HR/ML       NA    NA  NG/ML/MG    NA       NA   NA A0081216
  SITEID   SUBJID RAND                 TREATXT TRTCD PKCOLL PKBDFLD     PKTERM PERIODU PERIOD VISITU VISIT
1   1001 10011014   14 1X300 mg IR Fasted q12h     B  POINT  PLASMA PREGABALIN  PERIOD      2    DAY     4
2   1001 10011015   15 1X300 mg IR Fasted q12h     B  POINT  PLASMA PREGABALIN  PERIOD      2    DAY     4
3   1001 10011008    8 1X300 mg IR Fasted q12h     B  POINT  PLASMA PREGABALIN  PERIOD      1    DAY     4
4   1001 10011007    7 1X300 mg IR Fasted q12h     B  POINT  PLASMA PREGABALIN  PERIOD      2    DAY     4
                                    ACTTRT ACTTRTC         DOSEP DOSPND ROUTE RACES SEX  HT    WT HTUNI WTUNI TAU
1 PREGABALIN 1 X 300 MG IR CAPSULE FASTED        B PREGABALIN BD   DONE  ORAL WHITE   M 190 83.00    CM    KG  12
2 PREGABALIN 1 X 300 MG IR CAPSULE FASTED        B PREGABALIN BD   DONE  ORAL WHITE   M 172 69.10    CM    KG  12
3 PREGABALIN 1 X 300 MG IR CAPSULE FASTED        B PREGABALIN BD   DONE  ORAL WHITE   M 185 83.95    CM    KG  12
4 PREGABALIN 1 X 300 MG IR CAPSULE FASTED        B PREGABALIN BD   DONE  ORAL BLACK   M 171 67.45    CM    KG  12
 [ reached 'max' / getOption("max.print") -- omitted 14 rows ]
```

**Comment by jhhughes256** [comment]
This seems to be caused by CTROUGH2 and CTROUGHEND2 being NA. 

Looking inside ctrough.R this may be due to the use of the `&&` logical operator on line 130. This would give a single logical output, when it seems that a vector of logical outputs are desired as to filter tmp to create tmp_df.

```r
  tmp <- data.frame(time, conc)
  c_trough <- NULL
  
  # Find the pre-dose concentration value
  for(i in 1:length(tau)){
    start_time <- told[i]
    end_time <- start_time + tau[i]
    tmp_df <- tmp[start_time <= tmp$time && tmp$time <= end_time,]   ## this line
    if(is.null(c_trough)){
      c_trough <- tmp_df$conc[1]
    } else {
      c_trough <- c(c_trough, tmp_df$conc[1])
    }
  }
  return(c_trough)
```

When stepping through the function:

```r
Browse[2]> tmp
     time conc
1  11.917 2.29
2  12.333 2.32
3  12.667 9.47
4  13.017 9.54
5  13.500 8.53
6  14.000 8.24
7  16.000 5.63
8  18.000 4.11
9  21.000 3.04
10 24.000 2.20
Browse[2]> start_time <= tmp$time && tmp$time <= end_time
[1] FALSE
Browse[2]> start_time <= tmp$time & tmp$time <= end_time
 [1] FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
```

This error is expected to be triggered whenever the first value is prior to told.

**Comment by tensfeldt** [comment]
With commit b9591d0 tc011 M1SS parameterset=="PARAMETERDISPLAYLIST" and mct$TIME=="ACTUAL" CTROUGH2 is all NA. Note that there are numeric values for CTROUGH2 with mct$TIME=="NOMINAL"

**Comment by jhhughes256** [comment]
This issue still occurs for commit dce4ed8ecf.

**Comment by rudraya** [comment]
This issue should be resolved as of 1e16a39. Please let me know if it still occurs

**Comment by jhhughes256** [comment]
As of 1e16a39 CTROUGH2 is now correctly generated, causing PTROUGHR2 and PTROUGHREND2 to also be generated. Example of old and new behaviour below.

```r
> # 73826ef
> head(r_old, 4)
      SDEID DOSE1 DOSE2 DOSEC CMAX1 CMAX2  CMAXDN1  CMAXDN2 CMIN1 CMIN2 CLAST1 CLAST2 TMAX1  TMAX2 TMIN1  TMIN2 TLAST TLAG CTROUGH1
1  70745987   300   300   300  7460  6750 24.86667 22.50000  2310  2560   2680   2560 1.517 14.000     0 24.000    24    0     2310
2 130509330   300   300   300  8120  9540 27.06667 31.80000  1940  2200   2290   2200 1.500 13.017     0 24.000    24    0     1940
3 246409536   300   300   300  8130 12600 27.10000 42.00000   945  2240   2240   2420 0.667 12.667     0 11.917    24    0      945
4 249265153   300   300   300 12200 12100 40.66667 40.33333  1620  1720   2030   1720 1.000 13.000     0 24.000    24    0     1620
  CTROUGH2 CTROUGHEND1 CTROUGHEND2 PTROUGHR1 PTROUGHR2 PTROUGHREND1 PTROUGHREND2        KEL    KELC0 KELTMLO KELTMHI KELNOPT    KELRSQ
1       NA        2680          NA  3.229437        NA     2.783582           NA 0.10407221 31122.16      18      24       3 0.9999990
2       NA        2290          NA  4.185567        NA     3.545852           NA 0.10416094 26895.50      18      24       3 0.9995933
3       NA        2240          NA  8.603175        NA     3.629464           NA 0.09888024 25947.24      16      24       4 0.9990935
4       NA        2030          NA  7.530864        NA     6.009852           NA 0.13193783 40656.18      16      24       4 0.9993207
> # 1e16a39
> head(r_new, 4)
      SDEID DOSE1 DOSE2 DOSEC CMAX1 CMAX2  CMAXDN1  CMAXDN2 CMIN1 CMIN2 CLAST1 CLAST2 TMAX1  TMAX2 TMIN1  TMIN2 TLAST TLAG CTROUGH1
1  70745987   300   300   300  7460  6750 24.86667 22.50000  2310  2560   2680   2560 1.517 14.000     0 24.000    24    0     2310
2 130509330   300   300   300  8120  9540 27.06667 31.80000  1940  2200   2290   2200 1.500 13.017     0 24.000    24    0     1940
3 246409536   300   300   300  8130 12600 27.10000 42.00000   945  2240   2240   2420 0.667 12.667     0 11.917    24    0      945
4 249265153   300   300   300 12200 12100 40.66667 40.33333  1620  1720   2030   1720 1.000 13.000     0 24.000    24    0     1620
  CTROUGH2 CTROUGHEND1 CTROUGHEND2 PTROUGHR1 PTROUGHR2 PTROUGHREND1 PTROUGHREND2        KEL    KELC0 KELTMLO KELTMHI KELNOPT    KELRSQ
1     2580        2680        2560  3.229437  2.616279     2.783582     2.636719 0.10407221 31122.16      18      24       3 0.9999990
2     2320        2290        2200  4.185567  4.112069     3.545852     4.336364 0.10416094 26895.50      18      24       3 0.9995933
3     8540        2240        2420  8.603175  1.475410     3.629464     5.206612 0.09888024 25947.24      16      24       4 0.9990935
4     4210        2030        1720  7.530864  2.874109     6.009852     7.034884 0.13193783 40656.18      16      24       4 0.9993207
```
